### PR TITLE
feat(unix-cmd): User Experience improvements

### LINKED ISSUE #2643: feat(unix-cmd): User Experience improvements
**Is your feature request related to a problem? Please describe.**

There were some unexpected error messages when using the `Issue Unix Command` functionality.

- [ ] When no search path was entered, we get an error 
    - image
- [ ] When reviewing the history (terminal output), it would be nice to know which user and/or ssh-profile used to run the command
    - image
- [ ] When using it from the command palette, errors are not displayed if something goes wrong (no ssh profile, no username)
- [ ] When there are no ssh profiles, nothing happens
- [ ] When the ssh profile does not have user or password, we get an invalid username instead of being prompted for it
    - image

**Describe the solution you'd like**

These are the solutions I've considered for each item:
- _When no search path was entered..._
  - To be prompted for a path, even though it does not trigger a search (similar to using the Issue Unix Command function from the command palette)
- _When reviewing the history (terminal output)..._
  - To have the user and/or profile used to run the command
    ```
    CUST009 @ ssh-profile-1 : /u/users/cust009 > ls -al

    ...output...
    ```
- _When using it from the command palette..._
  -  To display a message when no ssh profile is present, or be prompted for credentials.
- _When there are no ssh profiles..._
  - To display a message when no ssh profile is present.
- _When the ssh profile does not have user or password..._
  - To be prompted for credentials 😋 

**Describe alternatives you've considered**

N/A

**Additional context**

Tested while reviewing:
- #2626

### PR DESCRIPTION
**Pull request by likhithanimma1**
## Proposed changes

 Fixes #2643 
 Enhanced Unix user experience, addressing feedback and improving usability.

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/command/UnixCommandHandler.ts
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/command/UnixCommandHandler.ts:138]: this text should be localized

File: packages/zowe-explorer/src/command/UnixCommandHandler.ts
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/command/UnixCommandHandler.ts:265]: This should also be localized

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
Thanks @likhithanimma1 for working on these polish items. I left a couple comments mainly on localization.  I was also wondering a couple things:
1. this doesn't seem to support using the base profile stored credentials with ssh profile even when they share host and ru values.
2. when prompted for ssh credentials are we planning to write them to file/store securely?  When I tested I was prompted but after checked config and no updates under ssh profile.

**Comment by likhithanimma1** [comment]
> Thanks @likhithanimma1 for working on these polish items. I left a couple comments mainly on localization. I was also wondering a couple things:
> 
> 1. this doesn't seem to support using the base profile stored credentials with ssh profile even when they share host and ru values.
> 2. when prompted for ssh credentials are we planning to write them to file/store securely?  When I tested I was prompted but after checked config and no updates under ssh profile.

Hey @JillieBeanSim !
1. This way of implementation will not support the stored credentials of the base profile with the ssh profile but if wanted can be modified to pick the username and password from the base profile for the default ssh profile. And if the base not containing them(user and pw) can be prompted for it.
2. We are not planning to write them to any file or store them billie!

**Comment by JillieBeanSim** [comment]
@likhithanimma1 we will need this to behave similarly as if using the team configuration file with cli commands for ssh.

**Review by t1m0thyj — COMMENTED** [review]
This looks pretty good, thanks @likhithanimma1!

Will re-review after merge conflicts have been resolved. A few questions:
* What is the expected behavior when the ssh profile does not have user or password? I'm seeing that nothing happens when I click "Issue Unix Command". But when my ssh profile has credentials, then it works as expected.
* For the prompt shown in the terminal output, it is currently formatted like this:
  `> ibmuser @ lpar1.ssh : ~ ls`
  Personally I think the spaces in between symbols look kind of weird. Would it make sense to use a format similar to most Linux terminals?
  `ibmuser@lpar1.ssh:~$ ls` OR `[ibmuser@lpar1.ssh ~]> ls`

**Review by traeok — COMMENTED** [review]
Changes LGTM, had a question about the use of the `profType` variable in `promptCredentials` 👍 will approve once conflicts are resolved

File: packages/zowe-explorer/src/Profiles.ts
Comment (likhithanimma1) [reply packages/zowe-explorer/src/Profiles.ts]: No, that would not be the intended behaviour. Just missed that case. Addressed the scenario and changed the prompt.

**Review by traeok — APPROVED** [review]
Tested and LGTM - thanks @likhithanimma1!

I like that the quick pick keeps track of your recent commands, this is a nice shortcut if you have to run the same command frequently 😋 

I filed #2728 as a possible future enhancement to allow adding/removing from this quick pick.

**Review by JillieBeanSim — APPROVED** [review]

**Pull request merged**
